// @teleporthq/teleport-lib-js/dist/lib/ElementsLibrary.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ElementsLibrary = /** @class */ (function () {
    function ElementsLibrary(libraryDefinition) {
        this.mappings = {};
        this.targets = {};
        var name = libraryDefinition.name, version = libraryDefinition.version, type = libraryDefinition.type, elements = libraryDefinition.elements;
        this.name = name;
        this.version = version;
        this.type = type;
        this.elements = elements;
    }
    /**
     * retrieves an element by it's name
     * @param name
     */
    ElementsLibrary.prototype.element = function (name) {
        return this.elements[name];
    };
    /**
     * adds a target mapping to the current library
     * sets up all internal wiring of dependencies between the library,
     * target and mapping
     * @param mapping
     */
    ElementsLibrary.prototype.useMapping = function (mapping) {
        this.mappings[mapping.name] = mapping;
        var target = mapping.target;
        this.targets[target.name] = target;
        mapping.setLibrary(this);
    };
    /**
     * retrieves a target for the current elements library
     * @param targetName
     */
    ElementsLibrary.prototype.target = function (targetName) {
        return this.targets[targetName];
    };
    return ElementsLibrary;
}());
exports.default = ElementsLibrary;
